filters/include/pcl/filters/experimental/voxel_grid.h

```diff
@@ -26,22 +26,22 @@ struct Voxel {
     Centroid() {}
     ~Centroid() {}
 
-    CentroidPoint<PointT> all_fields;
+    CentroidPoint<PointT>* all_fields;
     Eigen::Array4f xyz;
   };
 
   Voxel(const bool downsample_all_data) : downsample_all_data_(downsample_all_data)
   {
     num_pt_ = 0;
     if (downsample_all_data_)
-      centroid_.all_fields = CentroidPoint<PointT>();
+      centroid_.all_fields = new CentroidPoint<PointT>();
```

Comment:
Usage of heap doesn't seem right to me. If this is "solving" the segfault, I think the issue might be elsewhere

Reply:
It does solve the segfault but I have no clue why...